dev-libs/libmcrypt/libmcrypt-2.5.8.recipe

```diff
@@ -0,0 +1,68 @@
+SUMMARY="Replacements for the old Unix crypt"
+DESCRIPTION="libmcrypt are intended to be replacements for the \
+old Unix crypt, except that they are under the GPL and \
+support an ever-wider range of algorithms and modes."
+HOMEPAGE="http://mcrypt.sourceforge.net/"
+COPYRIGHT="1998-2002 Nikos Mavroyanopoulos"
+LICENSE="GNU LGPL v2.1"
+REVISION="1"
+SOURCE_URI="https://downloads.sourceforge.net/mcrypt/libmcrypt-$portVersion.tar.gz"
+CHECKSUM_SHA256="e4eb6c074bbab168ac47b947c195ff8cef9d51a211cdd18ca9c9ef34d27a373e"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion="4.4.8"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+PROVIDES="
+	libmcrypt$secondaryArchSuffix = $portVersion
+	lib:libmcrypt$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	libmcrypt${secondaryArchSuffix}_devel = $portVersion
+	devel:libmcrypt$secondaryArchSuffix = $portVersion
+	"
+REQUIRES_devel="
+	libmcrypt$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:aclocal
+	cmd:autoreconf
+	cmd:gcc$secondaryArchSuffix
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	"
+
+BUILD()
+{
+	autoreconf -vfi
+	runConfigure ./configure
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+
+	rm -f $libDir/*.la
+
+	prepareInstalledDevelLib libmcrypt
+	fixPkgconfig
+
+	packageEntries devel \
+		$developDir
```

Comment:
When you think your recipe is OK, you have to examine the contents of each hpkg (with `package list libmcrypt-2.5.8-1-x86_gcc2.hpkg` and `package list libmcrypt_devel-2.5.8-1-x86_gcc2.hpkg`).
With libmcrypt, we can see that :
* it has only one man page and that it is in section 3 (man3), so we can put $manDir in the devel subpackage. But since it builds on all architectures, we can drop it on secondary arch builds. Hence the `maybe_manDir` "trick".
* it has a `libmcrypt-config` in bin, so we need to move it to the devel subpackage. And because it's the only file in `$binDir`, whe can pass the whole `$binDir` tree to `packageEntries devel`.
* it has a `libmcrypt.m4` in `data/aclocal/` and that file is only needed in when you are compiling another program and wish to use autoconf to handle all the shared lib detection work. So you need to move that too to the devel subpackage.

So this is what we need:
```diff
 	prepareInstalledDevelLib libmcrypt
 	fixPkgconfig
 
+	if [ -n "$secondaryArchSuffix" ]; then
+		rm -rf $manDir
+		maybe_manDir=
+	else
+		maybe_manDir=$manDir
+	fi
+
 	packageEntries devel \
		$binDir \
		$dataDir \
-		$developDir
+		$developDir \
		$maybe_manDir
 }
```

Reply:
Oops, I forgot to insert the "+" in front of binDir and dataDir. (These are missing.)